### DISCUSSION TITLE
Is it possible to iterate through a vDataFrame

### DISCUSSION THREAD

**Discussion by emanuelh-cloud** [vertica/VerticaPy#1159] [Q&A] [answered]
Is it possible to iterate through a vDataFrame? Is there an option to perform iteration over the vDataFrame similar to how it is done with a Pandas DataFrame?

**Comment by oualib** [comment]
hi @emanuelh-cloud,
Thanks for your message.
Why would you like to do that? As ``vDataFrames`` do not work with indexes, iteration row by row will not make sense. Can you please describe your use case so we can see how we can help?

**Reply by emanuelh-cloud** [reply]
I would like to iterate the dataframe similar to pandas dataframe , for example using "iteritems()" and also  update the dataframe columns using ".at" similar to pandas.
Thank you

**Reply by oualib** [reply]
This functionality can be added only if we start supporting indexes, we need an internal discussion for that. Can you please describe exactly your use-case so we can see what we can do in our next ML meeting?

**Reply by emanuelh-cloud** [reply]
Hi @oualib, Many thanks for you response.
The scenario is straightforward: I want to incorporate functionality to iterate through the rows of a vDataframe and update a specific column based on the values of the preceding row.

**Reply by oualib** [reply, marked as answer]
@emanuelh-cloud this is the kind of operation which can only be done with indexes, it means that you are currently working with small datasets otherwise, Vertica will not handle many small transactions. 

We chose to not work with indexes. Except if it is a roadblock for a business use-case that you can share with our team, we can not guarantee the implementation of this feature.

Besides, your use-case can be solved with some quick VerticaPy code. If you want this type of solution. Give us an example of exactly what you want to realise and we can try to show you an example of a possible solution.
